src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,639 @@
+ */
+final class PhpServicesCreator
+{
+    private $factory;
+    private $creatorFactory;
+    private $stmts = [];
+    private $useStatements = [];
+
+    public function __construct()
+    {
+        $this->factory = new BuilderFactory();
+    }
+
+    public function convert(string $yamlFile): string
+    {
+        $this->creatorFactory = $this->factory->namespace(
+            'Symfony\Component\DependencyInjection\Loader\Configurator'
+        );
+
+        $closureStmt = $this->createClosureStmts((new Parser())->parse($yamlFile, 768));
+
+        $this->creatorFactory->addStmt(
+            new Node\Stmt\Return_(
+                new Node\Expr\Closure([
+                    'params' => [
+                        (new Param('configurator'))->setType('ContainerConfigurator')->getNode(),
+                    ],
+                    'stmts' => $closureStmt,
+                ])
+            )
+        );
+
+        $node = $this->creatorFactory->getNode();
+        $beforeClosure = !empty($this->useStatements) ? "\n" : null;
+        /*
+         * add a blank line between the last use statement and the closure
+         */
+        array_splice($node->stmts,
+            -1,
+            0,
+            [new Node\Name($beforeClosure."/*\n * This file is the entry point to configure your own services.\n */")]
+        );
+
+        return (new Standard())->prettyPrintFile([$node])."\n";
+    }
+
+    private function createClosureStmts(array $yamlArray): array
+    {
+        if (isset($yamlArray['parameters'])) {
+            $this->addNodeParameters($yamlArray['parameters']);
+        }
+
+        if (isset($yamlArray['imports'])) {
+            $this->addNodeImports($yamlArray['imports']);
+        }
+
+        if (isset($yamlArray['services'])) {
+            $this->addNodeServices($yamlArray['services']);
+        }
+
+        sort($this->useStatements);
+        /*
+         * Add at the end to sort them in alphabetical order.
+         */
+        foreach ($this->useStatements as $className) {
+            $this->creatorFactory->addStmt(
+                $this->factory->use($className)
+            );
+        }
+
+        // remove the last carriage return "\n" if exists.
+        $this->stmts[array_key_last($this->stmts)]->parts[0] = rtrim($this->stmts[array_key_last($this->stmts)]->parts[0], "\n");
+
+        return $this->stmts;
+    }
+
+    private function addNodeParameters(array $parameters): void
+    {
+        $this->addLineStmt('// Put parameters here that don\'t need to change on each machine where the app is deployed.');
+        $this->addLineStmt('// https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration.');
+
+        $this->addLineStmt('$parameters = $configurator->parameters();', true);
+
+        foreach ($parameters as $keyParameter => $values) {
+            $this->addLineStmt(
+                $this->createParameterCallMethod($keyParameter, $values), true
```

Comment:
`true` should be on the next line

Reply:
The logic for `createParameterCallMethod` is so small, it might be more clear to put it right inside this method. I'm normally a fan of extracting logic into private methods (even if it's not used multiple times) to add clarity. But here, the logic is so small - and `createParameterCallMethod()` itself isn't a very clear name (it's just hard to name) - that I think putting the logic right here and removing the private method is better.

Reply:
Done.